File: 1861-rotating-the-box/1861-rotating-the-box.py
class Solution:
    def rotateTheBox(self, boxGrid: List[List[str]]) -> List[List[str]]:
        rotated_box = [list(row) for row in zip(*boxGrid[::-1])]

        n, m = len(rotated_box) , len(rotated_box[0])

        for j in range(m):
            bottom = n - 1
            for i in range (n - 1, -1, -1):
                if rotated_box[i][j] == '*':
                    bottom = i - 1
                elif rotated_box[i][j] == '#':
                    if bottom != i:
                        rotated_box[bottom][j] = '#'
                        rotated_box[i][j] = '.'
                    bottom -= 1

        return rotated_box

File: 1861-rotating-the-box/1861-rotating-the-box.cpp
class Solution {
public:
    vector<vector<char>> rotateTheBox(vector<vector<char>>& boxGrid) {
        int m = boxGrid.size(), n = boxGrid[0].size();
        vector<vector<char>>rotatedBox(n, vector<char>(m));

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                rotatedBox[j][m - 1 - i] = boxGrid[i][j];
            }
        }

        for (int j = 0; j < m; j++) {
            int bottom = n - 1;
            for (int i = n - 1; i >= 0; i--) {
                if (rotatedBox[i][j] == '*') {
                    bottom = i - 1;
                }
                else if (rotatedBox[i][j] == '#') {
                    if (bottom != i) {
                        rotatedBox[bottom][j] = rotatedBox[i][j];
                        rotatedBox[i][j] = '.';
                    }
                    bottom--;
                }
            }
        }

        return rotatedBox;
    }
};